lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -47,16 +47,54 @@ defmodule BroadwayRabbitMQ.AmqpClient do
       type:
         {:or,
          [
+           :atom,
            {:custom, __MODULE__, :__validate_amqp_uri__, []},
            keyword_list: @connection_opts_schema
          ]},
-      default: [],
+      default: nil,
```

Comment:
In the spirit of "make illegal states not representable", I'm thinking I'd rather take an approach where we don't have a `:custom_pool` option at all, and instead we use

```elixir
connection: {:custom_pool, MyModule, init_args}
```

when using a custom pool. This would work well with the behaviour proposed in my other comment.

Reply:
i like the idea of passing the custom pool on the connection option. :+1: 
I'll do the behaviour.